firefox-ios/Storage/Rust/RustRemoteTabs.swift

```diff
@@ -175,183 +175,148 @@ public class RustRemoteTabs {
-        DispatchQueue.global().async { [unowned self] in
-            guard let commandQueue = self.commandQueue else {
+        queue.async { [unowned self] in
+            guard let tabsCommandQueue = self.tabsCommandQueue else {
                 let err = TabsApiError.UnexpectedTabsError(reason: "Command queue is not initialized") as MaybeErrorType
                 self.logger.log(err.description,
                                 level: .warning,
                                 category: .tabs)
                 return
             }
-            commandQueue
-                .removeRemoteCommand(deviceId: deviceId,
-                                     command: RemoteCommand.closeTab(url: url.absoluteString)) { result in
-                    self.handleCommandQueueErrors(result: result)
-                }
-        }
-    }
 
-    public func getUnsentCommandUrlsByDeviceId(deviceId: String, completion: @escaping ([String]) -> Void) {
-        DispatchQueue.global().async { [unowned self] in
-            self.getUnsentCommandsByDeviceId(deviceId: deviceId) { commands in
-                let urls = commands.map { item in
-                    switch item.command {
-                    case .closeTab(let url):
-                        return url
-                    }
-                }
-                completion(urls)
-            }
+            tabsCommandQueue
+                .removeRemoteCommand(deviceId: deviceId, command: RemoteCommand.closeTab(url: url.absoluteString))
         }
     }
 
-    public func setPendingCommandsSent(deviceId: String) {
-        DispatchQueue.global().async { [unowned self] in
-            guard let commandQueue = self.commandQueue else {
-                let err = TabsApiError.UnexpectedTabsError(reason: "Command queue is not initialized") as MaybeErrorType
-                self.logger.log(err.description,
-                                level: .warning,
-                                category: .tabs)
-                return
-            }
-
-            self.getUnsentCommandsByDeviceId(deviceId: deviceId) { commands in
-                commandQueue.setPendingCommandsSent(deviceId: deviceId, commands: commands) { errors in
-                    if errors.isEmpty {
-                        return
-                    } else {
-                        let errHeader = "Failed to set some pending commands as sent:\n"
-                        let errMessages: String = errors.reduce(errHeader, { result, err in
-                            result.appending("  - \(err.localizedDescription)\n")
-                        })
-                        self.logger.log(errMessages,
-                                        level: .warning,
-                                        category: .tabs)
-                    }
+    public func getUnsentCommandUrlsByDeviceId(deviceId: String, completion: @escaping ([String]) -> Void) {
+        self.getUnsentCommandsByDeviceId(deviceId: deviceId) { commands in
```

Comment:
Nit: `self` isn't necessary here AFAICS (typically we omit unless it's explicitly needed). I think there are a few other instances of this elsewhere also, but NAB.